src/streamlink/webbrowser/cdp/client.py

```diff
@@ -0,0 +1,406 @@
+        self._re_url: re.Pattern = self._url_pattern_to_regex_pattern(self.url_pattern)
+
+    def matches(self, request: fetch.RequestPaused) -> bool:
+        on_request: bool = request.response_status_code is None and request.response_error_reason is None
+        return on_request is self.on_request and self._re_url.match(request.request.url) is not None
+
+    @staticmethod
+    def _url_pattern_to_regex_pattern(url_pattern: str) -> re.Pattern:
+        pos = 0
+        regex = ""
+
+        for match in _re_url_pattern_wildcard.finditer(url_pattern):
+            regex += re.escape(match[1]) if match[1] else ""
+            if match[2]:
+                if len(match[2]) % 2:
+                    regex += f"{re.escape(match[2][:-1])}\\{match[3]}"
+                else:
+                    regex += re.escape(match[2])
+                    regex += ".+" if match[3] == "*" else "."
+            else:
+                regex += ".+" if match[3] == "*" else "."
+
+            pos = match.end()
+
+        regex += re.escape(url_pattern[pos:])
+
+        return re.compile(f"^{regex}$")
+
+
+@dataclass
+class CMRequestProxy:
+    body: str
+    response_code: int
+    response_headers: Optional[Mapping[str, str]]
+
+
+class CDPClient:
+    """
+    The public interface around :class:`ChromiumWebbrowser` and :class:`CDPConnection`.
+    It launches the web browser, initializes the CDP connection and provides the :meth:`session()` method
+    for creating a new :class:`CDPClientSession` that is tied to an empty new web browser tab.
+
+    :class:`CDPClientSession` provides a high-level API for intercepting network requests and responses,
+    as well as executing JavaScript expressions and retrieving async results.
+
+    Don't instantiate this class yourself, use the :meth:`CDPClient.launch()` async context manager classmethod.
+    """
+
+    def __init__(self, cdp_connection: CDPConnection, nursery: trio.Nursery):
+        self.cdp_connection = cdp_connection
+        self.nursery = nursery
+
+    @classmethod
+    def launch(
+        cls,
+        session: Streamlink,
+        runner: Callable[[Self], Coroutine],
+        executable: Optional[str] = None,
+        timeout: Optional[float] = None,
+        cdp_host: Optional[str] = None,
+        cdp_port: Optional[int] = None,
+        cdp_timeout: Optional[float] = None,
+        headless: Optional[bool] = None,
+    ) -> Any:
+        """
+        Start a new :mod:`trio` runloop and do the following things:
+
+        1. Launch the Chromium-based web browser using the provided parameters or respective session options
+        2. Initialize a new :class:`CDPConnection` and connect to the browser's remote debugging interface
+        3. Create a new :class:`CDPClient` instance
+        4. Execute the async runner callback with the :class:`CDPClient` instance as only argument
+
+        If the ``webbrowser`` session option is set to ``False``, then a :exc:`CDPError` will be raised.
```

Comment:
Might be some annoying comments about this because people are confused, should be added as a TODO for the release notes to clarify.

Reply:
I don't know what you mean. It's up to each plugin implementing the webbrowser API to handle errors. This is just the description of one of them.

Please take a look at my cdp-wip branch mentioned in the meta thread and check the Twitch plugin. Then you can play with the CLI args and see what happens. If it's disabled, then no client-integrity token will/can be retrieved. If this breaks the access token retrieval (which it currently does not), then the user will see why.